src/ansi.rs

```diff
@@ -678,6 +686,50 @@ bitflags! {
     }
 }
 
+/// This only applies to keys corresponding to ascii characters.
+///
+/// For the details on how to implement the mode handling correctly, consult [`XTerm's
+/// implementation`] and the output of XTerm's provided [`perl script`], the output of
+/// which for `CSI u` variant can be found [`here`]. Some libraries and implementations
+/// also use the [`fixterms`] definition of the `CSI u`.
+///
+/// The end escape sequence has a `CSI char; modifiers u` form while the original
+/// `CSI 27 ; modifier ; char ~`. The clients should prefer the `CSI u`, since it has
+/// more adoption.
+///
+/// [`XTerm's implementation`]: https://invisible-island.net/xterm/modified-keys.html
+/// [`perl script`]: https://github.com/ThomasDickey/xterm-snapshots/blob/master/vttests/modify-keys.pl
+/// [`here`]: https://github.com/alacritty/vte/blob/master/doc/modifyOtherKeys-example.txt
+/// [`fixterms`]: http://www.leonerd.org.uk/hacks/fixterms/
+#[repr(u8)]
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub enum ModifyOtherKeys {
+    /// Reset the state.
+    Reset,
+    /// Enables this feature except for keys with well-known behavior, e.g., Tab, Backspace and
+    /// some special control character cases which are built into the X11 library (e.g.,
+    /// Control-Space to make a NUL, or Control-3 to make an Escape character).
+    ///
+    /// Escape sequences shouldn't be emitted under the following circumstances:
+    ///
+    /// - When the key is in range of `[64;127]` and the modifier is either Control or Shift;
+    /// - when the key combination is a known control combination alias.
```

Comment:
```suggestion
    /// - When the key is in range of `[64;127]` and the modifier is either Control or Shift
    /// - When the key combination is a known control combination alias
```

It's a list, shouldn't be written like a sentence split apart with bullet points.

Reply:
depends, it's sort of common to make it a `;` separated in papers and have a e.g. dot in the last bullet point, but I don't really mind this thing in docs.